core/remote/index.js

```diff
@@ -404,13 +382,34 @@ class Remote /*:: implements Reader, Writer */ {
   }
 
   async findDocByPath(fpath /*: string */) /*: Promise<?MetadataRemoteInfo> */ {
-    const [dir, name] = dirAndName(fpath)
-    const { _id: dirID } = await this.remoteCozy.findDirectoryByPath(dir)
+    const [parentPath, name] = dirAndName(fpath)
+    const { _id: dirID } = await this.findDirectoryByPath(parentPath)
 
     const results = await this.remoteCozy.search({ dir_id: dirID, name })
     if (results.length > 0) return results[0]
   }
 
+  async findDirectoryByPath(
+    path /*: string */
+  ) /*: Promise<MetadataRemoteDir> */ {
+    // TODO: store a local copy of the root dir to limit requests to the Cozy?
+    if (path === '.') return this.remoteCozy.findDir(ROOT_DIR_ID)
```

Comment:
I don't understand why we need to get the root directory. Can't we recreate this document here? We know the id, we don't care of the revision, it is a folder, etc.

Reply:
I didn't want to hard code it but it's true that it's probably not gonna change much. And we would avoid quite some requests.